src/SlackConnector/SlackConnection.cs

```diff
@@ -63,20 +63,26 @@ public void Initialise(ConnectionInformation connectionInformation)
             ConnectedSince = DateTime.Now;
         }
 
-        private async Task ListenTo(InboundMessage inboundMessage)
+        private Task ListenTo(InboundMessage inboundMessage)
         {
-            if (inboundMessage?.MessageType != MessageType.Message)
-                return;
-            if (string.IsNullOrEmpty(inboundMessage.User))
-                return;
-            if (!string.IsNullOrEmpty(Self.Id) && inboundMessage.User == Self.Id)
-                return;
+            if (inboundMessage == null) Task.FromResult(false);
 
-            if (inboundMessage.Channel != null && !_connectedHubs.ContainsKey(inboundMessage.Channel))
+            switch (inboundMessage.MessageType)
```

Comment:
Could be, but I was not inclined to do it unless warranted. (And 3 events did not warrant it)

Also, the `switch` is much more efficient and readable, so I'd keep it even if there are 20 different message types to be handled.

If you can propose a design that would be acceptable to you, I'm open to it.